src/zeroband/training/config.py

```diff
@@ -3,67 +3,92 @@
 
 from pydantic import Field, model_validator
 
-from zeroband.utils.config import MultiMonitorConfig
-from zeroband.utils.models import AttnImpl
+from zeroband.utils.config import ModelConfig as BaseModelConfig
+from zeroband.utils.config import MultiMonitorConfig, PathConfig
+from zeroband.utils.models import AttnImplementation
 from zeroband.utils.pydantic_config import BaseConfig, BaseSettings
 
 
-class AdamConfig(BaseConfig):
+class ModelConfig(BaseModelConfig):
+    """Configures the model for training."""
+
+    attn: Annotated[
+        AttnImplementation, Field(default="flash_attention_2", description="The attention implementation to use.")
+    ]
+
+    compile: Annotated[
+        bool,
+        Field(
+            default=False,
+            description="Whether to compile the model using `torch.compile`. Currently discouraged because it was found to destabilize training.",
+        ),
+    ]
+
+
+class FSDPConfig(BaseConfig):
+    """Configures FSDP."""
+
+    reshard: Annotated[bool, Field(default=True, description="Whether to reshard the model after each forward pass.")]
```

Comment:
```suggestion
    reshard_after_forward: Annotated[bool, Field(default=True, description="Whether to reshard the model after each forward pass.")]
```

this is confusing imo

Reply:
we can support both with alias, the motivation for this was brevity